uast.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <assert.h>
+#include "uast.h"
+
+#define MAX_CHILDREN 100
+
+static xmlNodePtr recursiveConv(Node *node, xmlNodePtr parent)
```

Comment:
Missing header file for `xmlNodePtr` structure. We haven't talked about this, so this is just my opinion, but I personally prefer to have needed headers in each implementation file.